### ISSUE TITLE
A11y: Mailcow UI, Edit Mailbox: The toggle states of the Enforce TLS incoming / outgoing buttons are not exposed for screenreaders

### ISSUE THREAD

**Issue by robin24** [mailcow/mailcow-dockerized#4554]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/mailcow/mailcow-dockerized/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... I understand that not following the below instructions will result in immediate closure and/or deletion of my issue.
- [X] ... I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [X] ... I have understood that answers are voluntary and community-driven, and not commercial support.
- [X] ... I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

### Description

On the Edit Mailbox page of the Mailcow web UI, the toggle states of the "Enforce TLS outgoing" and "Enforce TLS incoming" buttons are not exposed to screenreaders.
For me as a blind screenreader user, this means I only hear an announcement of, for example, "Enforce TLS outgoing, button" without knowing whether this is currently on or off.

There's probably some visual treatment on these buttons to represent the current state, but no attribute that could be utilized by a screenreader to convey this appropriately.

### Logs

```bash
As this is purely a UI bug I can't think of any relevant logs to include here.
```

### Steps to reproduce

1. On a windows machine, download and install the free, open-source NVDA screenreader from https://www.nvaccess.org/download/.
Alternatively, on a Linux system with a graphical desktop the Orca screenreader can be installed from package repositories in most cases.
2. Open a browser (tested with Brave and Firefox in my case) and login to the Mailcow UI.
3. Open Configure > Mail Setup > Mailboxes and click the Edit link next to a mailbox.
4. Move screenreader focus to the "Enforce TLS incoming" and "Enforce TLS outgoing" buttons and observe speech output.

### System information

Again, this is probably irrelevant due to the nature of this bug - please advice if I'm wrong and I'll be happy to provide any missing details.

**Comment by mkuron** [comment]
Thanks for reporting, @robin24. Please try out the patch at #4556 and let me know whether it fixes the issue for you. I've followed the instructions at https://www.erikkroes.nl/blog/accessible-html-toggle-button-switch/ and used Apple's VoiceOver to confirm that the label changes from "Enforce TLS outgoing, button" to "Enforce TLS outgoing, off, switch". Hopefully Orca and NVDA will do something similar. Let us know if there are any other accessibility issues with Mailcow's web UI -- we don't systematically test for accessibility, but we haven't received many bug reports about it either.

**Comment by robin24** [comment]
Hey @mkuron,

Wow thanks for the awesome work, on a Sunday!

After copying the 2 changed files into my Mailcow installation and testing the edit mailbox page, I am happy to confirm your fix resolves the issue, this works perfectly now.

Also, really appreciate the encouragement to keep filing accessibility issues like that. I'm still new to Mailcow but have completed most of my setup by now, so I'm happy to report that screenreader accessibility is really good overall - these 2 toggles were the only ones I've found so far where the state was totally unclear to me.

Thanks again!

**Comment by DerLinkman** [comment]
Fixed with PR: https://github.com/mailcow/mailcow-dockerized/pull/4556
